langchain4j-core/src/main/java/dev/langchain4j/agent/tool/P.java

```diff
@@ -18,4 +18,10 @@
      * @return the description of a parameter
      */
     String value();
+
+    /**
+     * Whether the parameter is optional
+     * @return true if the parameter is optional, false otherwise
+     */
+    boolean optional() default false;
```

Comment:
Hmm I am wondering if this should be named `required` instead.
Reasoning: in the json schema to which this is eventually translated it is called `required`.

But I am not sure TBH, WDYT?

Reply:
I changed 'optional' to 'required'.
I defining it as 'optional' was just a habit with the boolean value defaulting to false. But I don't think it's important, like `org.springframework.web.bind.annotation.RequestParam`